pyerrors/fits.py

```diff
@@ -656,6 +690,189 @@ def chisqfunc_compact(d):
     return output
 
 
+def _combined_fit(x, y, func, silent=False, **kwargs):
+
+    if kwargs.get('correlated_fit') is True:
+        raise Exception("Correlated fit has not been implemented yet")
+
+    output = Fit_result()
+    output.fit_function = func
+
+    if kwargs.get('num_grad') is True:
+        jacobian = num_jacobian
+        hessian = num_hessian
+    else:
+        jacobian = auto_jacobian
+        hessian = auto_hessian
+
+    x_all = np.concatenate([np.array(o) for o in x.values()])
+    y_all = np.concatenate([np.array(o) for o in y.values()])
+
+    y_f = [o.value for o in y_all]
+    dy_f = [o.dvalue for o in y_all]
+
+    if len(x_all.shape) > 2:
+        raise Exception('Unknown format for x values')
+
+    # number of fit parameters
+    n_parms_ls = []
+    for key in func.keys():
+        if not callable(func[key]):
+            raise TypeError('func (key=' + key + ') is not a function.')
+        if len(x[key]) != len(y[key]):
+            raise Exception('x and y input (key=' + key + ') do not have the same length')
+        for i in range(42):
```

Comment:
This is also true for the other routines, but I just thought about it: Is 42 a good upper limit for the number of fit parameters? I guess so, but if you do combined fits, the probability that this is not the case is increased.

Reply:
This was more of a Douglas Adams reference than a meaningful upper bound. In practice, this should probably be more than enough but we could also raise it to 100 or so. I would expect the performance implications to be minimal.